Index: timeplot.js
===================================================================
--- timeplot.js	(revision 2054)
+++ timeplot.js	(working copy)
@@ -51,7 +51,8 @@
         roundValues:       ("roundValues" in params) ? params.roundValues : true,
         valuesOpacity:     ("valuesOpacity" in params) ? params.valuesOpacity : 75,
         bubbleWidth:       ("bubbleWidth" in params) ? params.bubbleWidth : 300,
-        bubbleHeight:      ("bubbleHeight" in params) ? params.bubbleHeight : 200
+        bubbleHeight:      ("bubbleHeight" in params) ? params.bubbleHeight : 200,
+	flat:              ("flat" in params) ? params.flat : false
     };
 };
 
Index: plot.js
===================================================================
--- plot.js	(revision 2054)
+++ plot.js	(working copy)
@@ -243,6 +243,7 @@
 
         ctx.lineWidth = this._plotInfo.lineWidth;
         ctx.lineJoin = 'miter';
+        var flat = this._plotInfo.flat;
 
         if (this._dataSource) {     
             if (this._plotInfo.fillColor) {
@@ -259,7 +260,12 @@
 
                 ctx.beginPath();
                 ctx.moveTo(0,0);
+                var y0 = 0;
                 this._plot(function(x,y) {
+                    if (flat) {
+                        ctx.lineTo(x,y0);
+                        y0=y;
+                    }
                     ctx.lineTo(x,y);
                 });
                 if (this._plotInfo.fillFrom == Number.NEGATIVE_INFINITY) {
@@ -278,11 +284,17 @@
                 ctx.strokeStyle = this._plotInfo.lineColor.toString();
                 ctx.beginPath();
                 var first = true;
+		var y0;
                 this._plot(function(x,y) {
                         if (first) {
                              first = false;
                              ctx.moveTo(x,y);
+			     y0 = y;
                         }
+			if (flat) {
+                          ctx.lineTo(x,y0);
+                          y0 = y;
+			}
                     ctx.lineTo(x,y);
                 });
                 ctx.stroke();
@@ -397,4 +409,4 @@
         }
     }
 
-}
\ No newline at end of file
+}
